### PR TITLE
Add jobs info

### LINKED ISSUE #552: Job endpoint enhancements for dashboard
1. [x] Store `origin` on job documents. Part of the job class; loaded during `/jobs/add` or `POST .../analyses/`.
2. [x]  Origin can be an optional `None` param on init; set to "system" or similar when not present. This is for jobs added by eg the rules settings.
3. [x] Add a `saved` string array to job object; this holds filenames of produced files. Set on engine & analysis uploads endpoints.
4. [ ] Save session information on analysis container references on the job.

cc @dpuccetti for frontend :thumbsup:

### PR DESCRIPTION
**Pull request by nagem**
Closes #552, #578.
### Review Checklist
- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

**Comment by nagem** [comment]
The `/jobs` endpoint doesn't return jobs run on analyses, only acquisitions. Let me know if you want that to change @dpuccetti.

File: api/jobs/jobs.py [hunk 1]
```diff
@@ -162,6 +173,16 @@ def insert(self):
         result = config.db.jobs.insert_one(self.mongo())
         return result.inserted_id
 
+    def save(self):
+        self.modified = datetime.datetime.utcnow()
+        update = self.mongo()
+        config.log.debug(update)
```
Comment (kofalt) [inline api/jobs/jobs.py]: Do we want this log statement in there when we merge?

File: api/jobs/jobs.py [hunk 2]
```diff
@@ -162,6 +173,16 @@ def insert(self):
...
+        job_id = update.pop('id')
```
Comment (kofalt) [inline api/jobs/jobs.py]: Is the id pop actually necessary?

File: api/jobs/jobs.py [hunk 3]
```diff
@@ -162,6 +173,16 @@ def insert(self):
...
+        result = config.db.jobs.replace_one({'_id': job_id}, update)
+        if result.modified_count != 1:
+            raise Exception('Job modification not saved')
+        return {'modified_count': 1}
```
Comment (kofalt) [inline api/jobs/jobs.py]: Why return a manual map rather than the actual result?

File: api/types.py [hunk 4]
```diff
@@ -5,5 +5,6 @@
     'user':    'user',    # An authenticated user
     'device':  'device',  # A connected device (reaper, script, etc)
     'job':     'job',     # Made on behalf of a job (downloading data, uploading results, etc)
-    'unknown': 'unknown', # Other or public
+    'system':  'system',  # Created by the system (a job initiated by a rule, etc)
+    'unknown': 'unknown'  # Other or public
```
Comment (kofalt) [inline api/types.py]: @dpuccetti Be aware that when this PR lands there will be a new origin type.

File: bin/database.py [hunk 5]
```diff
@@ -506,6 +507,20 @@ def upgrade_to_18():
...
+
+    Add origin information to job object
+    """
+
+    update = {
+        '$set': {
+            'origin' : {'type': str(Origin.unknown), 'id': None}
+        }
+    }
+    config.db.jobs.update_many({'origin.': {'$exists': False}}, update)
```
Comment (kofalt) [inline bin/database.py]: Do you need specifically `origin.`, or will `origin` work here?

File: api/jobs/jobs.py [hunk 6]
```diff
@@ -78,6 +87,8 @@ def __init__(self, name, inputs, destination=None, tags=None, attempt=1, previou
         self.id_             = id_
         self.config          = config_
         self.now             = now
+        self.origin          = origin
+        self.outputs         = []
```
Comment (kofalt) [inline api/jobs/jobs.py]: I hate to do this to you, but I think `outputs` should be renamed before this lands - `inputs` and `outputs` (currently a singleton `destination` but not forever) are kinda reserved meaning. I'd take `saved` or `result-filenames` instead?

File: bin/database.py (hunk 5 above)
Comment (nagem) [inline bin/database.py]: Good catch!

Comment (nagem) [inline bin/database.py]: But, surprisingly, `origin.` does work as intended.

File: api/jobs/jobs.py (hunk 3 above)
Comment (nagem) [inline api/jobs/jobs.py]: It's an existing practice across the API. Not married to the idea or anything, but might as well keep it consistent for now.

File: api/jobs/jobs.py (hunk 1 above)
Comment (nagem) [inline api/jobs/jobs.py]: Nope, thanks.

**Comment by nagem** [comment]
Made changes requested by @kofalt, merging.

**Pull request merged**